presto-tests/src/test/java/io/prestosql/connector/system/TestKillQuery.java

```diff
@@ -0,0 +1,111 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.connector.system;
+
+import io.prestosql.Session;
+import io.prestosql.plugin.tpch.TpchPlugin;
+import io.prestosql.spi.security.Identity;
+import io.prestosql.testing.AbstractTestQueryFramework;
+import io.prestosql.testing.DistributedQueryRunner;
+import io.prestosql.testing.QueryRunner;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.collect.MoreCollectors.toOptional;
+import static io.airlift.concurrent.Threads.threadsNamed;
+import static io.prestosql.testing.TestingAccessControlManager.TestingPrivilegeType.KILL_QUERY;
+import static io.prestosql.testing.TestingAccessControlManager.privilege;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.testng.Assert.assertFalse;
+
+@Test(singleThreaded = true)
+public class TestKillQuery
+        extends AbstractTestQueryFramework
+{
+    private final ExecutorService executor = Executors.newSingleThreadScheduledExecutor(threadsNamed(TestKillQuery.class.getSimpleName()));
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Session defaultSession = testSessionBuilder()
+                .setCatalog("tpch")
+                .setSchema("tiny")
+                .build();
+
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(defaultSession).build();
+        queryRunner.installPlugin(new TpchPlugin());
+        queryRunner.createCatalog("tpch", "tpch");
+        return queryRunner;
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        executor.shutdownNow();
+    }
+
+    @Test(timeOut = 60_000)
+    public void testKillQuery()
+            throws Exception
+    {
+        String testQueryId = "test_query_id_" + ThreadLocalRandom.current().nextInt(100_000_000);
+        Future<?> queryFuture = executor.submit(() -> {
+            getQueryRunner().execute(format("SELECT count(comment) as %s FROM tpch.sf100000.lineitem", testQueryId));
+        });
+
+        Optional<Object> queryIdValue = Optional.empty();
+        while (!queryIdValue.isPresent()) {
+            Thread.sleep(50);
```

Comment:
use failsafe?

Reply:
This test already has a timeout, so I don't think we need anything more.